### PR TITLE
update parent and consolidate on api plugin

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Comment by oleg-nenashev** [comment]
:bee:

File: pom.xml [hunk 1]
```diff
@@ -48,20 +54,4 @@
...
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-release-plugin</artifactId>
-                <version>2.5</version>
-                <configuration>
-                    <!-- https://jira.codehaus.org/browse/MRELEASE-271 workaround -->
-                    <preparationGoals>clean install</preparationGoals>
-                    <tagNameFormat>cloudbees-workflow-@{project.version}</tagNameFormat>
```
Comment (armfergom) [inline pom.xml:61]: Is it ok to remove this piece of configuration and just go with the default behaviour from parent pom? (`tagNameFormat`)

Comment (stephenc) [inline pom.xml:61]: that's just the tag in the git repo... and it was silly to include it anyway...

**Comment by stephenc** [comment]
@tfennelly WDYT?

File: pom.xml (hunk 1 above)
Comment (armfergom) [inline pom.xml:61]: fair enough :) :bee:

**Comment by armfergom** [comment]
:bee: sorry, I beed this in an inline comment

**Comment by andresrc** [comment]
:bee:

**Pull request merged**